code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,262 @@
+		)
+		// Stops postmarks from inheriting letter color.
+		// http://www.byond.com/docs/ref/#/atom/var/appearance_flags
+		SI.appearance_flags |= RESET_COLOR
+		add_overlay(SI)
+		bonus_stamp_offset -= 5
+
+	if(postmarked == TRUE)
+		var/image/PMI = image(
+			icon = icon,
+			icon_state = "postmark",
+			pixel_x = stamp_offset_x + rand(-3, 1),
+			pixel_y = stamp_offset_y + rand(bonus_stamp_offset + 3, 1)
+		)
+		PMI.appearance_flags |= RESET_COLOR
+		add_overlay(PMI)
+
+/obj/item/mail/attackby(obj/item/W, mob/user, params)
+	// Destination tagging
+	if(istype(W, /obj/item/dest_tagger))
+		var/obj/item/dest_tagger/O = W
+
+		if(sortTag != O.currTag)
+			var/tag = uppertext(GLOB.TAGGERLOCATIONS[O.currTag])
+			to_chat(user, "<span class='notice'>*[tag]*</span>")
+			sortTag = O.currTag
+			playsound(loc, 'sound/machines/twobeep_high.ogg', 100, TRUE)
+
+/obj/item/mail/attack_self(mob/user)
+	if(recipient && user != recipient)
+		to_chat(user, "<span class='notice'>You can't open somebody else's mail! That's <em>illegal</em>!</span>")
+		return
+
+	to_chat(user, "<span class='notice'>You start to unwrap the package...</span>")
+	if(!do_after(user, 15, target = user))
+		return
+	user.temporarilyRemoveItemFromInventory(src, TRUE)
+	unwrap_contents()
+	for(var/X in contents)
+		var/atom/movable/AM = X
+		user.put_in_hands(AM)
+	playsound(src.loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+	qdel(src)
+
+/obj/item/mail/proc/unwrap_contents()
+	for(var/obj/I in src.GetAllContents())
+		SEND_SIGNAL(I, COMSIG_STRUCTURE_UNWRAPPED)
+
+/// Accepts a mob to initialize goodies for a piece of mail.
+/obj/item/mail/proc/initialize_for_recipient(mob/new_recipient)
+	recipient = new_recipient
+	name = "[initial(name)] for [recipient.real_name] ([recipient.job])"
+	var/list/goodies = generic_goodies
+
+	var/datum/job/this_job = SSjob.name_occupations[recipient.job]
+	if(this_job)
+		if(this_job.paycheck_department && department_colors[this_job.paycheck_department])
+			color = department_colors[this_job.paycheck_department]
+		var/list/job_goodies = this_job.get_mail_goodies()
+		if(LAZYLEN(job_goodies))
+			// certain roles and jobs (prisoner) do not receive generic gifts.
+			if(this_job.exclusive_mail_goodies)
+				goodies = job_goodies
+			else
+				goodies += job_goodies
+
+	for(var/i = 0, i < goodie_count, i++)
+		var/T = pickweight(goodies)
+		if(ispath(T, /datum/reagent))
+			var/obj/item/reagent_containers/TI = new /obj/item/reagent_containers/glass/bottle(src)
+			TI.reagents.add_reagent(T, TI.volume)
+			TI.name = "[TI.reagents.reagent_list[1].name] bottle"
+			new_recipient.log_message("[key_name(new_recipient)] received reagent container [TI.name] in the mail ([T])", LOG_GAME)
+		else
+		//if(ispath(T, /obj))
+			var/atom/movable/TI = new T(src)
+			new_recipient.log_message("[key_name(new_recipient)] received [TI.name] in the mail ([T])", LOG_GAME)
+			//CRASH("[key_name(new_recipient)] received an unexpected type in the mail ([T])")
```

Comment:
Hello yes commented out code

Reply:
There was a weird issue where I wanted to check that path, but for some reason `ispath(T, /obj)` returns false when the goodies are stacks with defined amounts, i.e. `/obj/item/stack/sheet/mineral/plasma{amount = 5}` returns FALSE for that. I commented it out because I'd like a proper check instead of just assuming it must be an obj if it's not a reagent.

Reply:
using {} is generally considered bad practice, we use subtypes for that